### DISCUSSION TITLE
Using webhook for delivery digital product

### DISCUSSION THREAD

**Discussion by ryadhiprasetya** [swellstores/community#188] [Help] [answered]
Anyone can help me or any tutorial how to use webhook to delivery digital product after customer buy the product.

**Comment by ericingram** [comment]
Someone should build an app for this! I know various devs have implemented it

**Comment by sags95** [comment, marked as answer]
@ryadhiprasetya this might ultimately depend on what digital product you're selling and if you are using a dedicated DAM. Generally the steps would look like this:

1. Configure a webhook in Developer > Webhooks to fire on the `order.created` event. You can also select other events depending on the order flow.
2. In your webhook listener, depending if your store delivers a mix of both physical and digital goods, you would want to parse these orders. Digital products within orders would have `delivery: null`, so in the order payload you can inspect the `items` array for these items. If you only sell digital goods then this step is not necessary.
3. This last step will depend on what type of product you are selling. You would likely interact with the 3rd party app in order to fetch a licence key, or download, then provide access to the product. An easier way might also be through Zapier depending on what you're delivering and what platforms you're interacting with.
